javascript/ql/lib/semmle/javascript/frameworks/Spife.qll

```diff
@@ -0,0 +1,412 @@
+/**
+ * Provides classes for working with [Spife](https://github.com/npm/spife) applications.
+ */
+
+import javascript
+import semmle.javascript.frameworks.HTTP
+
+/**
+ * Provides classes for working with [Spife](https://github.com/npm/spife) applications.
+ */
+module Spife {
+  /**
+   * A call to a Spife method that sets up a route.
+   */
+  private class RouteSetup extends API::CallNode, Http::Servers::StandardRouteSetup {
+    TaggedTemplateExpr template;
+
+    RouteSetup() {
+      this.getCalleeNode().asExpr() = template and
+      API::moduleImport(["@npm/spife/routing", "spife/routing"])
+          .asSource()
+          .flowsToExpr(template.getTag())
+    }
+
+    private string getRoutePattern() {
+      // Concatenate the constant parts of the expression
+      result =
+        concat(Expr e, int i |
```

Comment:
```suggestion
        strictconcat(Expr e, int i |
```

If you just use `concat`, then you risk that `result = ""`, and that doesn't seem to be the thing you want.  

It makes no difference in the end (`hasLine` has the same result regardless), it's just a little nice-to-have.

Reply:
thanks